core/schema/directory.graphqls

```diff
@@ -37,4 +37,7 @@ type Directory {
 
     "The difference between this directory and an another directory"
     diff(other: DirectoryID!): Directory!
+
+    "Write the contents of the directory to a path on the host"
+    export(path: String!): Boolean!
```

Comment:
bikeshedding: if `export` is on `Directory`, why not have `import` (or `syncFromHost`) on `Directory` too? e.g. 
```graphql
type Directory {
  ...
  # merge the contents of hostPath from the host into destPath in the directory
  syncFromHost(hostPath: String!, destPath: String!, include: [String!], exclude: [String!]) Directory!
  ...
}
```

Just seems more intuitive, but if you and no one else agree please ignore.

Reply:
One thing to keep in mind is that we also plan to have `File.export` and `Container.export`.

Though to your point, I could see us adding `File.import` (load a single file into a `File`) and `Container.import` (load an OCI tarball into a `Container`).

@shykes @aluzzardi :bike: :derelict_house: :paintbrush:

Reply:
Thought about this a bit more, I think it's OK the way it is.

This was the dealbreaker: if we were to add a `Container.import` that reads from the host filesystem, people wouldn't be able to import from a `DirectoryID` (let's say you ran `nix build` or something to build an OCI tarball in Dagger). But if they can import from a `DirectoryID`, they can _also_ import from the host filesystem.

So it seems to me that it's an anti-goal for each object to support importing from the host filesystem, because they should ideally be reading from a `DirectoryID` for maximum flexibility instead.

I know this wasn't your argument, I'm just countering my own "logical conclusion." :)

Here are some supporting arguments for the current scheme:

1. Having `export` on each type feels more natural than having `exportX`, `X`, and `Z` on the host.
2. Having directories be "created" from their origin is already a common pattern, e.g. `Container.directory`
3. If we inverted it we might end up with `Directory.fromContainer` which feels like the type of pattern we're avoiding in step 1.

I don't know if I'm making sense here, there might be a better way to express it.